src/vi/goctruyentranhvui/src/eu/kanade/tachiyomi/extension/vi/goctruyentranhvui/GocTruyenTranhVui.kt

```diff
@@ -0,0 +1,181 @@
+package eu.kanade.tachiyomi.extension.vi.goctruyentranhvui
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import keiyoushi.utils.tryParse
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class GocTruyenTranhVui() : ParsedHttpSource() {
+    override val lang = "vi"
+
+    override val baseUrl = "https://goctruyentranhvui17.com"
+
+    override val name = "Goc Truyen Tranh Vui"
+
+    private val apiUrl = "$baseUrl/api/v2"
+
+    private val searchUrl = "$baseUrl/api/comic/search"
+
+    private val dateFormat = SimpleDateFormat("dd-MM-yyyy", Locale.US)
+
+    override val supportsLatest: Boolean = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val urls = client.newCall(GET(baseUrl + manga.url, headers)).execute().asJsoup()
+        val mangaId = checkChapterLists(urls)
```

Comment:
Store the manga's ID and URL in the `url` field (like `$id:$slug`) to avoid an extra network request. Follow other DTO suggestions from the previous PR reviews.

Reply:
> Store the manga's ID and URL in the `url` field (like `$id:$slug`) to avoid an extra network request. Follow other DTO suggestions from the previous PR reviews.

@stevenyomi You can review it now. It is now complete.